Lesson08-JS-V/homework/homework.js

```diff
@@ -0,0 +1,91 @@
+// Do not change any of the function names
+
+
+//PROBLEM 1 - VERSION 1
+//Error for both versions: createUserClass() › should return a user constructor that correctly builds user objects
+
+// create an ES6 class or a constructor function called User
+// it should accept an options object with username, name, email, and password properties
+// in the constructor set the username, name, email, and password properties
+// the constructor should have a method 'sayHi' on its prototype that returns the string 'Hello, my name is {{name}}'
+// {{name}} should be the name set on each instance
+// return the class
+function createUserClass() {
+    return User;
+}
+
+User.prototype.sayHi = function() {
+    return `Hello, my name is ${this.name}`;
+};
+
+
+function User (username, name, email, password) {
```

Comment:
I think  the `User` constructor expects a single object to be passed on that contains the 4 attributes. Something like:

```
 function User(options) {
    this.username = options.username;
    this.name = options.name;
    this.email = options.email;
    this.password = options.password;
  }
```